### ISSUE TITLE
melange/node/node_buffer.ml fromString binding invalid

### ISSUE THREAD

**Issue by jakubrpawlowski** [melange-re/melange#1236]
The order of the args in Buffer.fromString does not match javascript:

```
let test = Node.Buffer.fromString(~encoding=`utf8, "secret1234");
```

throws

```
Stack: TypeError [ERR_UNKNOWN_ENCODING]: Unknown encoding: secret1234: at fromString (node:buffer:481:13)
```

**Comment by jakubrpawlowski** [comment]
Here is an example binding that does work:

```
  [@mel.module "node:buffer"] [@mel.scope "Buffer"]
  external fromString: string => Node.Buffer.t = "from";
  ```

**Comment by davesnx** [comment]
Would you mind opening a PR with the fix? ^^

**Comment by anmonteiro** [comment]
thanks, indeed a bug. I fixed this in #1246 and added all the supported node.js encodings too.

there's also a new function `fromStringWithEncoding` if you do want to pass encoding explicitly, since we can't do `t -> ?foo -> 'a` without requiring `?foo` to always be applied.
